cli/app.go

```diff
@@ -1642,19 +1647,46 @@ After creation, use 'viam module update' to push your new module to app.viam.com
 					Name:  "generate",
 					Usage: "generate a new modular resource via prompts",
 					Flags: []cli.Flag{
+						&cli.StringFlag{
+							Name:  moduleFlagName,
+							Usage: "name to use for module",
+						},
 						&cli.StringFlag{
 							Name:  moduleFlagLanguage,
 							Usage: "language to use for module",
-							Value: "python",
+						},
+						&cli.BoolFlag{
+							Name:  moduleFlagIsPublic,
+							Usage: "set module to public",
 						},
 						&cli.StringFlag{
-							Name:  moduleFlagResourceType,
-							Usage: "resource type to use in module",
+							Name:  moduleFlagPublicNamespace,
+							Usage: "namespace or organization ID of module",
 						},
 						&cli.StringFlag{
 							Name:  moduleFlagResourceSubtype,
 							Usage: "resource subtype to use in module",
 						},
+						&cli.StringFlag{
+							Name:  moduleFlagResourceType,
+							Usage: "resource type to use in module",
+						},
+						&cli.StringFlag{
+							Name:  moduleFlagModelName,
+							Usage: "resource model name to use in module",
+						},
+						&cli.BoolFlag{
+							Name:  moduleFlagEnableCloud,
+							Usage: "generate Github workflows to build module",
+						},
+						&cli.BoolFlag{
+							Name:  moduleFlagRegister,
+							Usage: "register module with Viam to associate with your organization",
+						},
+						&cli.BoolFlag{
+							Name:  moduleFlagTest,
+							Usage: "don't do organization or namespace check",
```

Comment:
The check that's being skipped is `resolveOrg`, which gets the orgID while also checking if the user is part of that organization. I called it `test` since the only time people would use this is if they were testing the module while putting in a fake namespace, similar to your `do most of the upfront work`. I'll try to rename these to make it more clear.

Reply:
Got it, thanks! Sorry to nitpick but can we rename to `dry-run`? It's not a big deal either way but I think `dry-run` is more precise in terms of indicating what this flag does. And/or, do we expect/want users to use this flag at all? If not, we should probably hide the flag by setting `Hidden`.

Reply:
I thought it would be helpful for the users to be able to experiment with the module generator, but I guess this isn't necessary. I didn't know a setting `Hidden` existed. I'll use it!

Reply:
sg! I defer to your judgment on whether you think this is useful to expose to users :)